### PR TITLE
[cisco_ios]: escape hyphen inside regex char list

### PR DESCRIPTION
**Pull request by pkoutsovasilis**
## Proposed commit message

After tackling [this issue](https://github.com/elastic/integrations/issues/9489), a respective [PR](https://github.com/elastic/elastic-package/pull/1755) was created to `elastic-package` to mainstream the detection of similar warnings. Thus this PR fixes the following similar to the former issue regex-related warning for `cisco_ios`
```
character class has '-' without escape
```

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

N/A 

## Related issues

N/A

## Screenshots

N/A

### CODE REVIEW

File: packages/cisco_ios/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -42,7 +42,7 @@ processors:
         CISCO_PRIORITY_MSGCOUNT: '<%{NONNEGINT:log.syslog.priority:long}>(?:%{NONNEGINT:cisco.ios.message_count})?(?:: )?'
         CISCO_TIMESTAMP: '[*]?%{CISCOTIMESTAMP:_temp_.cisco_timestamp}(?: %{CISCO_TZ:_temp_.tz})?'
         CISCO_UPTIME: '[0-9a-zA-Z]+'
-        CISCO_HOSTNAME: '[a-zA-Z][0-9a-zA-Z-_]{0,61}[0-9a-zA-Z]?'
+        CISCO_HOSTNAME: '[a-zA-Z][0-9a-zA-Z\-_]{0,61}[0-9a-zA-Z]?'
```
Comment (jrmolin) [inline packages/cisco_ios/data_stream/log/elasticsearch/ingest_pipeline/default.yml:45]: should this be `\\-` like the other PR?

**Review by jrmolin — APPROVED** [review]
one question, not going to block for an answer.

**Comment by dwhyrock** [comment]
From the report:

Data stream | Previous EPS | New EPS | Diff (%) | Result
----------- | ------------ | ------- | -------- | ------
`log` | 2207.51 | 545.85 | -1661.66 (-75.27%) | :broken_heart:

Is EPS "events per second"?  And should we be concerned with the numbers here?

File: packages/cisco_ios/data_stream/log/elasticsearch/ingest_pipeline/default.yml (hunk 1 above)
Comment (pkoutsovasilis) [reply packages/cisco_ios/data_stream/log/elasticsearch/ingest_pipeline/default.yml:45]: here this pattern is with `'` not `"` so the `\` doesn't need escaping

**Comment by pkoutsovasilis** [comment]
> From the report:
> 
> Data stream	Previous EPS	New EPS	Diff (%)	Result
> `log`	2207.51	545.85	-1661.66 (-75.27%)	💔
> Is EPS "events per second"? And should we be concerned with the numbers here?

hmmm not sure to be honest I just escaped a hyphen I didn't expect such a massive impact on perf 🤷‍♂️

**Comment by taylor-swanson** [comment]
I pulled the changes locally and ran benchmarks myself. I'm not noticing any notable differences between the current version in main and this one. If anything the timing seems slightly tighter and ever so slightly faster with this change than main. Unfortunately, the benchmark tool doesn't give anything more specific for stats (percentiles, standard deviation, etc) so that's all based on "feel".

I'm not sure exactly how the CI runs the benchmarks, but if it's only once, then unfortunately there's a chance that it has a bad run and show a large "change", when in reality it could run it again and show that it's the same or faster.

**Comment by pkoutsovasilis** [comment]
So @taylor-swanson @dwhyrock @jrmolin, after having a look at the CI steps, the elastic-package benchmark is invoked only once [here](https://github.com/elastic/integrations/blob/main/.buildkite/scripts/common.sh#L955). The baseline benchmark results are uploaded from main, so when a commit is merged the affected packages are detected and for ones found, benchmark is run and pushed in the respective buildkite artifact. That said, I did the same practice as Taylor in his comments and my local running benchmark report is not exhibiting such a substantial perf difference. However, I did push a commit that revert the fix and just does a dummy change to cause the benchmarking to execute and the performance hindering is gone 🤷‍♂️ Any more ideas?

**Comment by pkoutsovasilis** [comment]
> ### 🚀 Benchmarks report
> #### Package `cisco_ios` 👍(1) 💚(0) 💔(0)
> Expand to view
> Data stream	Previous EPS	New EPS	Diff (%)	Result
> `log`	0	637.76	637.76 ( - %)	👍



I think somehow all my tweaking and messing around managed to mess up with the benchmarking. In an attempt to recover that, I am gonna close this one and open a new one with my experiments

**Pull request closed without merging**